.github/workflows/update-gcs-bucket.yaml

```diff
@@ -9,39 +9,62 @@ on:
 
 env:
   node_version: 16
+  typesense_url: https://typesense.camino.network
+  default_collection_name: camino-docs-default
 
 jobs:
-  BuildAndDeploy:
+  build_deploy_scrape:
+    name: Build, Deploy & Scrape
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
-      - uses: actions/setup-node@v3
-        with:
-          node-version: ${{ env.node_version }}
-          
-      - name: install yarn dependencies        
-        run: yarn install
 
-      - name: build
-        run: yarn build
-
-      - name: set bucket names
-        id: setBucketNames
+      - name: Prepare variables
+        id: prepareVariables
         run: |
           if [[ "${{ github.ref_name }}" == "main" ]]; then
               echo "bucket=docs.camino.network" >> $GITHUB_OUTPUT
+              echo "collection=camino-docs-`date +%s`" >> $GITHUB_OUTPUT
+              echo "loadbalancer=docs-lb" >> $GITHUB_OUTPUT
           elif [[ "${{ github.ref_name }}" == "dev" ]]; then
               echo "bucket=dev.docs.camino.network" >> $GITHUB_OUTPUT
+              echo "collection=camino-docs-dev-`date +%s`" >> $GITHUB_OUTPUT
+              echo "loadbalancer=dev-docs-lb" >> $GITHUB_OUTPUT
           fi
 
-      - name: 'gcloud authentication'
-        id: 'auth'
+      - uses: actions/setup-node@v3
+        with:
+          node-version: ${{ env.node_version }}
+          
+      - name: Install yarn dependencies        
+        run: yarn install
+
+      - name: Set collection name for Docusaurus
+        run: sed -i "s/camino-docs-default/${{ steps.prepareVariables.outputs.collection }}/g" docusaurus.config.js
+
+      - name: Build
+        run: yarn build
+
+      - name: Cloud authentication
+        id: auth
         uses: 'google-github-actions/auth@v1'
         with:
           credentials_json: '${{ secrets.GOOGLE_CREDENTIALS }}'
 
-      - name: 'Set up Cloud SDK'
-        uses: 'google-github-actions/setup-gcloud@v1'
+      - name: Set up Cloud SDK
+        uses: google-github-actions/setup-gcloud@v1
+
+      - name: Update bucket
+        run: |
+          gsutil -m rsync -R -d build gs://${{ steps.prepareVariables.outputs.bucket }}
 
-      - name: update bucket files
-        run: gsutil rsync -R -d build gs://${{ steps.setBucketNames.outputs.bucket }}
+      - name: Serve local site for the scraper
+        run: docker run -d -p80:80 -v$PWD/build:/usr/share/nginx/html nginx:stable-alpine
+
+      - name: Run local scraping and push to TypeSense
```

Comment:
As mentioned in DM: I suggest running scraper first, then gcs.
If now scraper fails, a doc without search will be deployed

Reply:
Done & tested